### PR TITLE
backports: for v0.4.0-beta.1

### PR DESCRIPTION
**Pull request by rsmitty**
This PR bundles the backports for release-0.4 branch needed for v0.4.0-beta.1

See:

65043b7cf85407e9e4683dbda1d96083d9e1de45
e0041f6567a61a388f5e69769fe28924e4f4fe7f

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,8 +61,76 @@ You will need at least the upstream CAPI components, the Talos bootstrap provide
...
+      generateType: controlplane
+```
+
+Note you must provide an infrastructure template for your control plane.
+See your infrastructure provider for how to craft that.
+
+Note the generateType mentioned above.
+This is a required value in the spec for both controlplane and worker ("join") nodes.
+For a no-frills control plane config, you can simply specify `controlplane` depending on each config section.
+When creating a TalosControlPlane this way, you can then retrieve the talosconfig file that allows for osctl interaction with your nodes by doing something like `kubectl get talosconfig -o yaml talos-cp-xxxx -o jsonpath='{.status.talosConfig}'` after creation.
```
Comment (frezbo) [inline README.md:103]:
```suggestion
When creating a TalosControlPlane this way, you can then retrieve the talosconfig file that allows for talosctl interaction with your nodes by doing something like `kubectl get talosconfig -o yaml talos-cp-xxxx -o jsonpath='{.status.talosConfig}'` after creation.
```

File: README.md [hunk 2]
```diff
@@ -58,8 +61,76 @@ You will need at least the upstream CAPI components, the Talos bootstrap provide
...
+
+A basic config:
+
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: TalosControlPlane
+metadata:
+  name: talos-cp
+spec:
+  version: v1.18.1
```
Comment (frezbo) [inline README.md:86]: shouldn't the example show a valid talos version?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-Instead, it is now issues bootstrap and keeps bootstrap status information in the `TalosControlPlane` resource.
-Bootstrap state can also be retrieved from the conditions.
+This release of CACPPT brings compatibility with CAPI v1beta1.
 
 
-### CAPI v1beta1
+### Init Node Deprecation
 
-This release of CACPPT brings compatibility with CAPI v1beta1.
+This release of CACPPT deprecates the `init` node that was used in previous Talos Linux designs. Specifying only a `controlplane` as part of the spec is now all that's required and the provider itself will handle automatic bootstrap of the first controlplane node.
```
Comment (frezbo) [inline CHANGELOG.md:18]: I guess we can also mention which version of talos started deprecating `init` config

File: hack/release.toml [hunk 4]
```diff
@@ -21,10 +21,8 @@ preface = """\
...
-        title = "`init` nodes deprecation"
+    [notes.init]
+        title = "Init Node Deprecation"
         description = """\
-Starting from this release CACPPT no longer relies on init nodes to bootstrap the cluster.
-Instead, it is now issues bootstrap and keeps bootstrap status information in the `TalosControlPlane` resource.
-Bootstrap state can also be retrieved from the conditions.
-"""
+This release of CACPPT deprecates the `init` node that was used in previous Talos Linux designs. Specifying only a `controlplane` as part of the spec is now all that's required and the provider itself will handle automatic bootstrap of the first controlplane node.
+"""
```
Comment (frezbo) [inline hack/release.toml:28]: nit: missing newline

**Review by frezbo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,23 +1,21 @@
-## [CAPI Control Plane Provider Talos 0.4.0-beta.0](https://github.com/talos-systems/cluster-api-control-plane-provider-talos/releases/tag/v0.4.0-beta.0) (2021-11-29)
+## [CAPI Control Plane Provider Talos 0.4.0-beta.1](https://github.com/talos-systems/cluster-api-control-plane-provider-talos/releases/tag/v0.4.0-beta.1) (2021-12-09)
 
-Welcome to the v0.4.0-beta.0 release of CAPI Control Plane Provider Talos!  
+Welcome to the v0.4.0-beta.1 release of CAPI Control Plane Provider Talos!
```
Comment (smira) [inline CHANGELOG.md:3]: We had beta.0 release, right? So we should be merging changelogs, not replacing them?

**Review by smira — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

**Pull request merged**